Cython/Utility/CythonFunction.c

```diff
@@ -362,6 +367,36 @@ __Pyx_CyFunction_get_annotations(__pyx_CyFunctionObject *op, CYTHON_UNUSED void
     return result;
 }
 
+static PyObject *
+__Pyx_CyFunction_get_is_coroutine(__pyx_CyFunctionObject *op, CYTHON_UNUSED void *context) {
+#if PY_VERSION_HEX >= 0x030503C1
+    // on v3.5.3rc1 the marker object was introduced
+    if (!op->func_is_coroutine) {
+        if (op->flags & __Pyx_CYFUNCTION_COROUTINE) {
+            PyObject *module, *fromlist, *marker = PYIDENT("_is_coroutine");
+            fromlist = PyList_New(1);
+            if (unlikely(!fromlist)) goto ignore;
+            PyList_SET_ITEM(fromlist, 0, marker);
+            module = PyImport_ImportModuleLevelObject(PYIDENT("asyncio.coroutines"), NULL, NULL, fromlist, 0);
+            Py_DECREF(fromlist);
+            if (unlikely(!module)) goto ignore;
+            op->func_is_coroutine = __Pyx_PyObject_GetAttrStr(module, marker);
+            Py_DECREF(module);
+            if (unlikely(!op->func_is_coroutine)) goto ignore;
+            return op->func_is_coroutine;
```

Comment:
@scoder if I understood, GetAttrStr already owns the reference, so we can't Py_INCREF it. Is this correct?

Reply:
We are dealing with two references here: one that we cache, and one that we return.
```suggestion
            return __Pyx_NewRef(op->func_is_coroutine);
```